src/history/hs_rec.c

```diff
@@ -197,9 +197,19 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         if (!hs_read_all_flag)
             F_CLR(cursor, WT_CURSTD_HS_READ_ALL);
     }
-    if (ret == 0)
+    if (ret == 0) {
+        /*
+         * Fail the eviction if we detect out of order timestamp when checkpoint is running. We
+         * cannot modify the history store to fix the out of order timestamp updates as it may make
+         * the history store checkpoint inconsistent.
+         */
+        if (checkpoint_running) {
+            ret = EBUSY;
```

Comment:
Add statistic for this eviction failure case.

Reply:
Added in 7cacea7